spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -430,6 +432,20 @@ public void setFileExistsMode(FileExistsMode fileExistsMode) {
 		}
 	}
 
+	public void setChmodOctal(String chmod) {
```

Comment:
M-m-m. Can't we really rely here on the Java octal representation?
E.g. the `600` from your test-case should be changed just to `0600`.